cloud/aws/metrics/build-scraper

```diff
@@ -2,8 +2,23 @@
 
 # Builds the prometheus metrics scraper Docker image
 
-docker build \
-  -t docker.io/civiform/aws-metrics-scraper:latest \
-  -f metrics_scraper.Dockerfile .
+readonly SHORT_SHA="$(git rev-parse --short HEAD)"
+readonly DATE_IN_UNIX_SECONDS="$(date +%s)"
+readonly SNAPSHOT_TAG="SNAPSHOT-${SHORT_SHA}-${DATE_IN_UNIX_SECONDS}"
+readonly IMAGE="aws-metrics-scraper"
+
+PLATFORM_ARG=()
+if [[ "${PLATFORM}" ]]; then
+  PLATFORM_ARG=(--platform "${PLATFORM}")
+fi
 
-docker push docker.io/civiform/aws-metrics-scraper:latest
+set -e
+set +x
+
+echo "start ${IMAGE} build"
+docker buildx create --use
+docker buildx build \
+  "${PLATFORM_ARG[@]}" --push \
+  -t docker.io/civiform/${IMAGE}:latest \
+  -t docker.io/civiform/${IMAGE}:${SNAPSHOT_TAG} \
+  -f metrics_scraper.Dockerfile .
```

Comment:
**suggestion (blocking):** please give https://google.github.io/styleguide/shellguide.html a read. Some specific things throughout this file:
- double quote all strings containing variables
- case distinction between constants and changing vars
- use explicit checks in conditionals

Reply:
Thanks! Updated.